### ISSUE TITLE
How to multiple server-side renders?

### ISSUE THREAD

**Issue by blakeembrey** [blakeembrey/react-free-style#1]
I assume the style cache right now is just going to keep growing. How do we automatically set up the react element so it'll reset the cache on each render? How can we use multiple caches for each render?

**Comment by blakeembrey** [comment]
Component based styles (fixes the third party module use case):

```js
var Style = require('react-free-style').component()
var React = require('react')

module.exports = React.createClass({
  render: function () {
    return <div><Style.Element /></div>
  }
})
```

It requires that the style element is always added to the element itself. Is there a way to automatically hook onto render? With this, `Style.Element` will automatically switch between a `<style />` and an empty node to avoid multiple style tags in the output. Only the first render will hold the styles. If the first one is removed, another needs to take over the dependency.

**Comment by blakeembrey** [comment]
Using contexts?

```js
// app.jsx
var Style = require('react-free-style')
var React = require('react')

module.exports = React.createClass({
  mixin: [Style.Context],
  render: function () {
    return <div><Child /><Style.Element /></div>
  }
})

// child.jsx
var Style = require('react-free-style').group()
var React = require('react')

var STYLE = Style.registerStyle({
  color: 'red'
})

module.exports = React.createClass({
  mixin: [Style.Mixin],
  componentWillMount: function () {
    this.registerStyle(Style)

    this.backgroundColor = this.registerStyle({ backgroundColor: 'blue' })
  },
  render: function () {
    return <div className={Style.join(STYLE.className, this.backgroundColor.className)}>
  }
})
```

**Comment by blakeembrey** [comment]
Solved with `v0.3`.
